From e4cf5d1a98469c17133f33fc8220e6b41f65e9a0 Mon Sep 17 00:00:00 2001
From: David Tardon <dtardon@redhat.com>
Date: Mon, 8 Dec 2014 18:06:01 +0100
Subject: [PATCH] don't copy stylesheet's HelpID if this's a rename

This causes problems for renamed copies of std. styles, as they have
special handling in SdStyleSheet::SetHelpId, which would set the API
name to std. value too. That means that there would be two (or more)
stylesheets with the same API name. On export, only one of them would be
exported.

Change-Id: I19d243bd3f73995e8de2d251e3d8b7f93219c3a9
---
 sd/source/core/stlpool.cxx | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/sd/source/core/stlpool.cxx b/sd/source/core/stlpool.cxx
index 095054e..94f8f96 100644
--- a/sd/source/core/stlpool.cxx
+++ b/sd/source/core/stlpool.cxx
@@ -699,7 +699,8 @@ void SdStyleSheetPool::CopySheets(SdStyleSheetPool& rSourcePool, SfxStyleFamily
             if( !aParent.isEmpty() )
                 aNewStyles.push_back( std::pair< rtl::Reference< SfxStyleSheetBase >, OUString >( xNewSheet, aParent ) );
 
-            xNewSheet->SetHelpId( aHelpFile, xSheet->GetHelpId( aHelpFile ) );
+            if( !bAddToList )
+                xNewSheet->SetHelpId( aHelpFile, xSheet->GetHelpId( aHelpFile ) );
             xNewSheet->GetItemSet().Put( xSheet->GetItemSet() );
 
             rCreatedSheets.push_back( SdStyleSheetRef( static_cast< SdStyleSheet* >( xNewSheet.get() ) ) );
-- 
2.1.0

